### ISSUE TITLE
Got LockException error during inference large-scale datasets

### ISSUE THREAD

**Issue by coldmanck** [facebookresearch/detectron2#4089]
I've modified `plain_train_net.py` to inference my larger-scale dataset which is around 2.6 million images with pre-trained object detection models. When I run with smaller number of data (say, 1000 imgs), the code works well. However, when I ran with my full dataset it encountered an error, `portalocker.exceptions.LockException: [Errno 11] Resource temporarily unavailable`. Full error message is as follows. 

```
Traceback (most recent call last):
  File "plain_train_net.py", line 507, in <module>
    args=(params,),
  File "/home/tiger/anaconda/lib/python3.7/site-packages/detectron2/engine/launch.py", line 79, in launch
    daemon=False,
  File "/home/tiger/anaconda/lib/python3.7/site-packages/torch/multiprocessing/spawn.py", line 230, in spawn
    return start_processes(fn, args, nprocs, join, daemon, start_method='spawn')
  File "/home/tiger/anaconda/lib/python3.7/site-packages/torch/multiprocessing/spawn.py", line 188, in start_processes
    while not context.join():
  File "/home/tiger/anaconda/lib/python3.7/site-packages/torch/multiprocessing/spawn.py", line 150, in join
    raise ProcessRaisedException(msg, error_index, failed_process.pid)
torch.multiprocessing.spawn.ProcessRaisedException: 

-- Process 4 terminated with the following error:
Traceback (most recent call last):
  File "/home/tiger/anaconda/lib/python3.7/site-packages/torch/multiprocessing/spawn.py", line 59, in _wrap
    fn(i, *args)
  File "/home/tiger/anaconda/lib/python3.7/site-packages/detectron2/engine/launch.py", line 126, in _distributed_worker
    main_func(*args)
  File "/opt/tiger/my_project/detectron2/projects/TTLiveSTAD/plain_train_net.py", line 468, in main
    return do_test(cfg, model)
  File "/opt/tiger/my_project/detectron2/projects/TTLiveSTAD/plain_train_net.py", line 257, in do_test
    cfg, dataset_name, os.path.join(cfg.OUTPUT_DIR, "pred_person_proposals", f'cls_{cfg.TTLIVE_DATA.CLS}_{cfg.TTLIVE_DATA.PART}')
  File "/opt/tiger/my_project/detectron2/projects/TTLiveSTAD/plain_train_net.py", line 230, in get_evaluator
    evaluator_list.append(COCOPersonEvaluator(dataset_name, cfg, output_dir=output_folder))
  File "/opt/tiger/my_project/detectron2/projects/TTLiveSTAD/evaluation/evaluator.py", line 46, in __init__
    super().__init__(dataset_name, output_dir=output_dir, distributed=distributed)
  File "/home/tiger/anaconda/lib/python3.7/site-packages/detectron2/evaluation/coco_evaluation.py", line 130, in __init__
    convert_to_coco_json(dataset_name, cache_path)
  File "/home/tiger/anaconda/lib/python3.7/site-packages/detectron2/data/datasets/coco.py", line 462, in convert_to_coco_json
    with file_lock(output_file):
  File "/home/tiger/anaconda/lib/python3.7/site-packages/portalocker/utils.py", line 157, in __enter__
    return self.acquire()
  File "/home/tiger/anaconda/lib/python3.7/site-packages/portalocker/utils.py", line 272, in acquire
    raise exceptions.LockException(exception)
portalocker.exceptions.LockException: [Errno 11] Resource temporarily unavailable
```

This happened during the step of **Trying to convert 'my_dataset' to COCO format ...**

Basically, I do not understand what does this error message means, especially for my (larger-scale) dataset. I found that this error happened around **60 mins after** the above prompt **Trying to convert 'my_dataset' to COCO format ...** logged on my screen. I did some simple searches and I found that it might because of the `timeout` set for portalocker.Lock() which is `3600` in [Facebook's iopath](https://github.com/facebookresearch/iopath/blob/4e20ced4058009fe12687f5b89820b759bd3ce83/iopath/common/file_io.py#L98). So I suspect that it means I got a timeout error where one of my threads did something with the file more than one hour. If that's the case, isn't this an unreasonable choice? What should I fix my issue? Thank you very much!

**Comment by coldmanck** [comment]
@ppwwyyxx  I sincerely hope that you and the developers can have a look at my issue. Thank you :)

**Comment by coldmanck** [comment]
Follow-up: my issue was solved by implementing my own `file_lock()` function which modifies [this line](https://github.com/facebookresearch/iopath/blob/4e20ced4058009fe12687f5b89820b759bd3ce83/iopath/common/file_io.py#L98) by increasing the `timeout` to `60*60*48` which means 48 hours, as my dataset needs to process for around 24 hours. I think maybe we shouldn't set a timeout (or, for example, set `timeout=float('inf')`).

**Comment by ppwwyyxx** [comment]
You're right about the issue relating to timeout. We didn't anticipate that the dataset generation can take more than an hour.

It sounds like a good solution is to make the timeout of `file_lock` an argument, and call it with a larger timeout value (or infinite, if possible) from detectron2.

**Comment by coldmanck** [comment]
> You're right about the issue relating to timeout. We didn't anticipate that the dataset generation can take more than an hour.
> 
> It sounds like a good solution is to make the timeout of `file_lock` an argument, and call it with a larger timeout value (or infinite, if possible) from detectron2.

Thanks for the reply. Yes, I do think the `timeout=3600` value should be configurable.

**Comment by Shreyz-max** [comment]
If no one is working on it, can I make a PR @ppwwyyxx @coldmanck ?
